### PR TITLE
check if quiz_id is non-negative integer

### PR DESCRIPTION
**Pull request by yulheekim**

### CODE REVIEW

File: resources/bought.py [hunk 1]
```diff
@@ -36,12 +36,11 @@ class Bought(Resource):
 
     def post(self, mode):
         data = Bought.parser.parse_args()
-        try:
-            quiz_id = int(mode)
-        except:
-            return "Param should be convertable to in for post", 400
 
-        error_message, status, response = BoughtController.make_bought(data['name'], data['email'], quiz_id, data['q1'], data['q2'], data['q3'])
+        if not BoughtController.check_quiz_id(mode):
```
Comment (jameswhang) [inline resources/bought.py]: do you really need a function call here? Can't you just do the type checking in here instead?

File: controllers/notbought.py [hunk 2]
```diff
@@ -23,6 +23,10 @@ def make_notbought(cls, name, quiz_id, q1, q2, q3):
 
         return "", 201, result
 
+    @classmethod
```
Comment (jameswhang) [inline controllers/notbought.py]: You're adding in 2 layers of check_quiz_id for what reason exactly..?

File: controllers/bought.py [hunk 3]
```diff
@@ -16,6 +16,10 @@ def make_bought(cls, name, email, quiz_id, q1, q2, q3):
 
         return "", 201, None
 
+    @classmethod
```
Comment (jameswhang) [inline controllers/bought.py]: Why is this necessary

**Comment by jameswhang** [comment]
Also pls add in test for these kind of changes to verify your code works.

File: resources/notbought.py [hunk 4]
```diff
@@ -33,12 +33,14 @@ class NotBought(Resource):
...
-        try:
-            quiz_id = int(mode)
-        except:
-            return "Param needs to be an integer for post.", 400
 
-        error_message, status, response = NotBoughtController.make_notbought(data['name'], quiz_id, data['q1'], data['q2'], data['q3'])
+        if not mode.isdigit():
+            return {"error_message": "Param should be convertible to int (non-negative) for post"}, 400
+
+        if len(mode) >= 2:
```
Comment (Mark-Jung) [inline resources/notbought.py]: Why do you check from this if statement? I think the purpose of this was to get rid of the 0s in front of the number, such as 03 or 003 to mean 3. However, the int casting on line 43 already takes care of this case.
screen shot 2018-06-20 at 12 19 03 pm

File: tests/test_notbought.py [hunk 5]
```diff
@@ -65,14 +65,32 @@ def test_notbought_post_get(self):
...
         helper.print_error(new_notbought, 201)
         # check if the card arrives
         resp_data = json.loads(new_notbought.data.decode())
         self.assertEqual(resp_data['response']['Cards'][0]['name'], 'name1')
         self.assertEqual(resp_data['response']['Cards'][1]['name'], 'name2')
         self.assertEqual(resp_data['response']['Cards'][2]['name'], 'name3')
+
+        new_notbought = helper.make_notbought(self, 'name1', 'name2', 'name3', 'san', "03")
+        helper.print_error(new_notbought, 201)
+        resp_data = json.loads(new_notbought.data.decode())
```
Comment (Mark-Jung) [inline tests/test_notbought.py]: I don't think line 79~82 is necessary. We already check if the correct card information arrives in line 73 to 75.

**Comment by jameswhang** [comment]
Pls take care of merge conflict as well.

File: resources/notbought.py (hunk 4 above)
Comment (yulheekim) [reply resources/notbought.py]: Good point. We checked this because we tested with 03, not "03" as a string. It's fixed now.

**Comment by jameswhang** [comment]
lgtm. feel free to merge it
